ross/api_report

```diff
@@ -505,10 +516,161 @@ class Report:
 
         Parameters
         ----------
-        (Check what we need to calculate the applied cross coupling and list
-         them as parameters)
+        machine_type : str
+            Choose the rotating machine type
+            Options: "compressor", "turbine" or "axial flow"
+        D : list
+            Impeller diameter, m (in.),
+            Blade pitch diameter, m (in.),
+        H : list
+            Minimum diffuser width per impeller, m (in.),
+            Effective blade height, m (in.),
+        HP : list
+            Rated power per stage/impeller, W (HP),
+        N : float
+            Operating speed, rpm,
+        RHOd : float
+            Discharge gas density per impeller, kg/m3 (lbm/in.3),
+        RHOs : float
+            Suction gas density per impeller, kg/m3 (lbm/in.3).
+        steps : int
+            Number of steps in the range.
+            Default is 11.
+        unit : str
+            Adopted unit system.
+            Default is "m"
+
+        Return
+        ------
+
+        Example
+        -------
+
         """
-        pass
+        if unit == "m":
+            C = 9.55
+        elif unit == "in":
+            C = 63.0
+        else:
+            raise TypeError("choose between meters (m) or inches (in)")
+
+        if len(D) != len(H):
+            raise Exception("length of D must be the same of H")
+
+        # Qa - Antecipated cross-coupling - API 684 - SP6.8.5.6
+        if machine_type == "compressor":
+            Bc = 3.
+            Dc, Hc = D, H
+            Qa = 0.
+            for i in range(len(self.rotor.disk_elements)):
+                qi = (HP[i] * Bc * C * RHOd) / (Dc[i] * Hc[i] * N * RHOs)
+                Qa = Qa + qi
+
+        # Qa - Antecipated cross-coupling - API 684 - SP6.8.5.6
+        if machine_type == "turbine" or machine_type == "axial flow":
+            Bt = 1.5
+            Dt, Ht = D, H
+            Qa = 0.
+            for i in range(len(self.rotor.disk_elements)):
+                qi = (HP[i] * Bt * C) / (Dt[i] * Ht[i] * N)
+                Qa = Qa + qi
+
+        bearings = [copy(b) for b in self.rotor.bearing_seal_elements]
+        for brg in self.rotor.bearing_seal_elements:
+            cross_coupling = BearingElement(n=brg.n, kxx=0, cxx=0, kxy=Qa, kyx=-Qa)
```

Comment:
I don't think the anticipated cross-coupling should be applied to the bearing position.
Can you verify that?

Reply:
You are right. I'm fixing it right now and making some other improvements. It's based on if the rotor is between bearings or overhung, and the cross-coupling is introduced at rotor's mid-span or at the center of gravity of the stage or impeller for single overhung rotors.